variants/audit-logging/template.rb

```diff
@@ -0,0 +1,39 @@
+
+source_paths.unshift(File.dirname(__FILE__))
+
+copy_file("variants/audit-logging/lib/auditor_log_formatter.rb", "lib/auditor_log_formatter.rb")
+copy_file("variants/audit-logging/services/auditor.rb", "app/services/auditor.rb")
+directory("variants/audit-logging/services/auditor", "app/services/auditor")
+
+insert_into_file "app/controllers/application_controller.rb", after: /^  end/ do
+  <<-'RUBY'
+
+
+    ##
+    # Example of generating an audit log event for "Thing A"
+    #
+    # You can edit the shape of audit log events as your app requires.
+    #
+    def note_thing_a_in_audit_log
+      event = Auditor::Events::ThingA.new(
+        current_user: current_user,
+        remote_ip: request.remote_ip,
+        event_info: { anything: "anything" } # event_info is a Hash of whatever else you need to store about the event
+      )
+      Auditor.log(event)
+    end
```

Comment:
Is there another/better way for me to leave clues in the code for how to use the audit logging?

Reply:
What about _usage_ of the logger? 😄 

In the long-run I definitely want to try and get us compliant with standards like NZISM which say we need to record stuff like user login events - imo that would double as a nice example of how to use audit logging.

If we make this variant always enabled, then we could have other variants (like the Devise one) include setting up logging events

Reply:
> If we make this variant always enabled, then we could have other variants (like the Devise one) include setting up logging events
Oh I like that.